File: 01gpio/061ultra_sonic/ultra_sonic_wpi.py
import wiringpi as wpi, time
PIN_TRIG = 18
PIN_ECHO = 23

wpi.wiringPiSetupGpio()
wpi.pinMode(PIN_TRIG, wpi.OUTPUT)
wpi.pinMode(PIN_ECHO, wpi.INPUT)

while True:
	wpi.digitalWrite(PIN_TRIG, wpi.HIGH)
	wpi.delayMicroseconds(10)
	wpi.digitalWrite(PIN_TRIG, wpi.LOW)
	while wpi.digitalRead(PIN_ECHO) == 0:
		pass
	start_time = time.time()
	while wpi.digitalRead(PIN_ECHO) == 1:
		pass
	travel_time = time.time() - start_time
	distance = travel_time * 17160
	print('Distance: %dCm '%round(distance, 2)  , travel_time)
	wpi.delay(10)

File: 01gpio/061ultra_sonic/ultra_sonic_wpi.js
var wpi = require('wiring-pi');
var pin_trig = 18;
var pin_echo = 23;
var start_time = end_time = travel_time =  distance = 0;

wpi.wiringPiSetupGpio();
wpi.pinMode(pin_trig, wpi.OUTPUT);
wpi.pinMode(pin_echo, wpi.INPUT);

while(true){
	wpi.digitalWrite(pin_trig, wpi.LOW);
	wpi.delay(200);
	wpi.digitalWrite(pin_trig, wpi.HIGH);
	wpi.delayMicroseconds(10);
	wpi.digitalWrite(pin_trig, wpi.LOW);
	while(wpi.digitalRead(pin_echo) === 0){
		start_time = process.hrtime();
	}
	while(wpi.digitalRead(pin_echo) === 1){
		end_time = process.hrtime(start_time);
	}
	travel_time = end_time[0] + (end_time[1] * 1e-9);
	distance = travel_time * 17160;
	console.log(distance.toFixed(2) + 'cm ' + travel_time);
}